proto/osmosis/pool-incentives/v1beta1/incentives.proto

```diff
@@ -0,0 +1,48 @@
+syntax = "proto3";
+package osmosis.poolincentives.v1beta1;
+
+import "gogoproto/gogo.proto";
+import "google/protobuf/duration.proto";
+
+option go_package = "github.com/c-osmosis/osmosis/x/pool-incentives/types";
+
+message Params {
+  option (gogoproto.goproto_stringer) = false;
+
+  // minted_denom is the denomination of the coin expected to be minted by the minting module.
+  // Pool-incentives module doesn’t actually mint the coin itself, but rather manages the distribution of coins that matches the defined minted_denom.
+  string minted_denom = 1 [(gogoproto.moretags) = "yaml:\"minted_denom\""];
+  // allocation_ratio defines the proportion of the minted minted_denom that is to be allocated as pool incentives.
+  string allocation_ratio = 2 [
+    (gogoproto.customtype) = "github.com/cosmos/cosmos-sdk/types.Dec",
+    (gogoproto.moretags) = "yaml:\"allocation_ratio\"",
+    (gogoproto.nullable) = false
+  ];
+}
+
+message LockableDurationsInfo {
+  repeated google.protobuf.Duration lockable_durations = 1 [
+    (gogoproto.nullable)    = false,
+    (gogoproto.stdduration) = true,
+    (gogoproto.moretags)    = "yaml:\"lockable_durations\""
+  ];
+}
+
+message DistrInfo {
+  string total_weight = 1 [
+    (gogoproto.customtype) = "github.com/cosmos/cosmos-sdk/types.Int",
+    (gogoproto.moretags) = "yaml:\"total_weight\"",
+    (gogoproto.nullable) = false
+  ];
+  repeated DistrRecord records = 2 [(gogoproto.nullable) = false];
```

Comment:
TotalWeight should be kept separately though DistrRecords have all information for total-weights?

Reply:
Yeah, we need to keep track of the sum.  Don't want to iterate over all the DistrRecords every time we want the sum.